src/test/mock-utils.ts

```diff
@@ -0,0 +1,20 @@
+// NOTE: This file is referenced as a pathGroup pattern in .eslintrc (import/order)
+
+import fs from 'fs';
+
+// Declare originalFs outside of mockReadFileSync to prevent infinite recursion errors in mockReadFileSync.
+const originalFs = fs.readFileSync;
+
+/**
+ * Mocks the fs library if the file path includes the specified file path substring
+ * and otherwise returns the original content.
+ */
+export const mockReadFileSync = (filePathSubstr: string, mockValue: string) => {
+  return jest.spyOn(fs, 'readFileSync').mockImplementationOnce((...args) => {
```

Comment:
This (unforunately) needs to be mockImplementation because the mock implementation will get consumed if you try to log something before the file is loaded.

I've checked codesearch (https://cs.github.com/api3dao/airseeker/blob/ee2bf2db512c0e69915281af6daca2406d910b9f/src/config.test.ts?q=%22const+mockReadFileSync%22#L12) and we have the same problem in airseeker.

Reply:
Oh right, I think we had something similar in Airnode :sweat_smile: